metrics/black_list.go

```diff
@@ -0,0 +1,72 @@
+package metrics
+
+import (
+	"regexp"
+	"strings"
+)
+
+// BlackList encapsulates the logic needed to filter based on a string.
+type BlackList struct {
+	list        map[string]struct{}
+	rList       []*regexp.Regexp
+}
+
+// New constructs a new BlackList based on a white- and a
+// blacklist. Only one of them can be not empty.
+func New(b map[string]struct{}) (*BlackList, error) {
```

Comment:
New is a bit unclear as the package consists of more code than just BlackList.

Reply:
Hi,iwankgb, Thanks for your advisce,I have changed New to NewDenyList.